hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsV_0_38Module.java

```diff
@@ -0,0 +1,274 @@
+import static org.hyperledger.besu.evm.MainnetEVMs.registerShanghaiOperations;
+import static org.hyperledger.besu.evm.operation.SStoreOperation.FRONTIER_MINIMUM;
+
+import com.hedera.node.app.service.evm.contracts.operations.HederaBalanceOperation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaDelegateCallOperation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaEvmChainIdOperation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaEvmCreate2Operation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaEvmCreateOperation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaExtCodeCopyOperation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaExtCodeHashOperation;
+import com.hedera.node.app.service.evm.contracts.operations.HederaExtCodeSizeOperation;
+import com.hedera.node.app.service.mono.context.TransactionContext;
+import com.hedera.node.app.service.mono.context.properties.GlobalDynamicProperties;
+import com.hedera.node.app.service.mono.contracts.ContractsModule.V_0_38;
+import com.hedera.node.app.service.mono.contracts.operation.HederaCallCodeOperation;
+import com.hedera.node.app.service.mono.contracts.operation.HederaCallOperationV034;
+import com.hedera.node.app.service.mono.contracts.operation.HederaLogOperation;
+import com.hedera.node.app.service.mono.contracts.operation.HederaPrngSeedOperator;
+import com.hedera.node.app.service.mono.contracts.operation.HederaSLoadOperation;
+import com.hedera.node.app.service.mono.contracts.operation.HederaSStoreOperation;
+import com.hedera.node.app.service.mono.contracts.operation.HederaSelfDestructOperation;
+import com.hedera.node.app.service.mono.contracts.operation.HederaStaticCallOperation;
+import com.hedera.node.app.service.mono.contracts.sources.EvmSigsVerifier;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import dagger.multibindings.IntoSet;
+import java.math.BigInteger;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiPredicate;
+import java.util.stream.Collectors;
+import javax.inject.Singleton;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.EvmSpecVersion;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.internal.EvmConfiguration;
+import org.hyperledger.besu.evm.operation.Operation;
+import org.hyperledger.besu.evm.operation.OperationRegistry;
+import org.hyperledger.besu.evm.precompile.MainnetPrecompiledContracts;
+import org.hyperledger.besu.evm.precompile.PrecompileContractRegistry;
+import org.hyperledger.besu.evm.precompile.PrecompiledContract;
+
+@Module
+public interface ContractsV_0_38Module {
+
+    String EVM_VERSION_0_38 = "v0.38";
+
+    @Provides
+    @Singleton
+    @V_0_38
+    static BiPredicate<Address, MessageFrame> provideAddressValidator(
```

Comment:
This was pre-existing work, as you said it was already duplicated in v0.30 and v0.34.  

We could factor it out short term, but it will come back just as soon as we have versions with different precompiled contract sets.  The duplication is a emergent property of using dagger.

Delegating it to another (shared) method would further erode the readability of these injectors.